### PR TITLE
V2.0.0 kowhai update config

### PR DESCRIPTION
**Pull request by red010b37**
This branch sets about auto loading and reloading the config from the file system. 

This branch also refactors and makes a clear distinction from user and server config.  The configs are now accessed directly by each package instead of pass throughs making them self healing as configs update

The old daemon server is removed in favor of a single server application system - this reduces complexity later on as we add more services

### CODE REVIEW

File: app/conf/userConf.go [hunk 1]
```diff
@@ -6,18 +6,30 @@ import (
...
+		for range ticker.C {
+			LoadUserConfig()
+		}
+	}()
+}
+
+
 // LoadUserConfig loads the config from a file
-func LoadUserConfig() (*Config, error)  {
+func LoadUserConfig() (error)  {
```
Comment (matt-auckland) [inline app/conf/userConf.go]: Is `error` meant to be `err`? Doesn't look like `error` is used at all

File: app/conf/serverConf.go [hunk 2]
```diff
@@ -2,20 +2,21 @@ package conf
...
 	ReleaseAPI string
+	DaemonHeartbeat int64
 }
 
 
 
 
 // LoadUserConfig loads the config from a file
-func LoadServerConfig() (*ServerConfig, error)  {
+func LoadServerConfig() (ServerConfig, error)  {
```
Comment (matt-auckland) [inline app/conf/serverConf.go]: another `error` and `err` situation

File: app/daemon/daemon.go [hunk 3]
```diff
@@ -100,17 +101,91 @@ type GitHubReleaseData struct {
...
+	if err != nil {
+		isLiving = false
+	}
+
+	return isLiving
+
+}
+
+
+func DownloadAndStart(serverConfig conf.ServerConfig, userConfig conf.UserConfig) (*exec.Cmd, error) {
```
Comment (matt-auckland) [inline app/daemon/daemon.go]: whats with the double brackets? () ()

File: app/conf/serverConf.go (hunk 2 above)
Comment (red010b37) [reply app/conf/serverConf.go]: this is defining the return types, error is the types, and err is the var  :)

File: app/conf/userConf.go (hunk 1 above)
Comment (red010b37) [reply app/conf/userConf.go]: error is the one of the return types :)

File: app/daemon/daemon.go (hunk 3 above)
Comment (red010b37) [reply app/daemon/daemon.go]: first brackets is the function params that will be passed in, the second is the return types that will be passed back

File: app/conf/userConf.go (hunk 1 above)
Comment (red010b37) [reply app/conf/userConf.go]: this is defining the return types, error is the types, and err is the var :)

File: app/conf/serverConf.go (hunk 2 above)
Comment (matt-auckland) [reply app/conf/serverConf.go]: riiiight, that makes a lot of sense.

**Pull request merged**